packages/material-ui/src/TablePagination/TablePagination.d.ts

```diff
@@ -69,6 +75,18 @@ export interface TablePaginationTypeMap<P, D extends React.ElementType> {
        * }
        */
       getItemAriaLabel?: (type: 'first' | 'last' | 'next' | 'previous') => string;
+      /**
+       * Customize how the `from` and `to` row range is calculated. Invoked with a `{ page, rowsPerPage, count }`
+       * object.
+       *
+       * @default function defaultCalculateRowRange({ page, rowsPerPage, count }) {
+       *   return {
+       *     from: count === 0 ? 0 : page * rowsPerPage + 1,
+       *     to: count !== -1 ? Math.min(count, (page + 1) * rowsPerPage) : (page + 1) * rowsPerPage
+       *   };
+       * }
+       */
+      calculateRowRange?: (paginationInfo: CalculateRowRangeArgs) => React.ReactNode;
```

Comment:
`getRowRange`?

Reply:
If we were to move ahead, very happy to update this with your preferred nomenclature.